rules/REQUEST-921-PROTOCOL-ATTACK.conf

```diff
@@ -24,14 +24,14 @@ SecRule TX:EXECUTING_PARANOIA_LEVEL "@lt 1" "id:921012,phase:2,pass,nolog,skipAf
 # -=[ HTTP Request Smuggling ]=-
 #
 # [ Rule Logic ]
-# This rule looks for a CR/LF character in combination with a HTTP / WEBDAV method name.
+# This rule looks for a HTTP / WEBDAV method name in combination with the word http/\d or a CR/LF character.
 # This would point to an attempt to inject a 2nd request into the request, thus bypassing
 # tests carried out on the primary request.
 #
 # [ References ]
 # http://projects.webappsec.org/HTTP-Request-Smuggling
 #
-SecRule ARGS_NAMES|ARGS|XML:/* "@rx [\n\r]+(?:get|post|head|options|connect|put|delete|trace|track|patch|propfind|propatch|mkcol|copy|move|lock|unlock)\s+[^\s]+(?:\s+http|[\r\n])" \
+SecRule ARGS_NAMES|ARGS|REQUEST_BODY|XML:/* "@rx (?:get|post|head|options|connect|put|delete|trace|track|patch|propfind|propatch|mkcol|copy|move|lock|unlock)\s+[^\s]+(?:\s+http\/\d|[\r\n])" \
```

Comment:
I think this modification is not complete, a part of the new regex is missing after `unlock)`: `(?:\/|\w)`. The line in 3.3 is this:

```
SecRule ARGS_NAMES|ARGS|REQUEST_BODY|XML:/* "@rx (?:get|post|head|options|connect|put|delete|trace|track|patch|propfind|propatch|mkcol|copy|move|lock|unlock)\s+(?:\/|\w)[^\s]*(?:\s+http\/\d|[\r\n])" \
```